src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacReorderValidator.java

```diff
@@ -0,0 +1,151 @@
+ *
+ * You may obtain a copy of the License at
+ * 		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.cam.cl.dtg.isaac.quiz;
+
+import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.isaac.dos.IsaacReorderQuestion;
+import uk.ac.cam.cl.dtg.isaac.dos.QuestionValidationResponse;
+import uk.ac.cam.cl.dtg.isaac.dos.content.Choice;
+import uk.ac.cam.cl.dtg.isaac.dos.content.Content;
+import uk.ac.cam.cl.dtg.isaac.dos.content.ContentBase;
+import uk.ac.cam.cl.dtg.isaac.dos.content.Item;
+import uk.ac.cam.cl.dtg.isaac.dos.content.ItemChoice;
+import uk.ac.cam.cl.dtg.isaac.dos.content.Question;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Validator that provides functionality to validate reorder questions. It is essentially a copy of the cloze question
+ * validator.
+ */
+public class IsaacReorderValidator implements IValidator {
+    private static final Logger log = LoggerFactory.getLogger(IsaacReorderValidator.class);
+
+    @Override
+    public final QuestionValidationResponse validateQuestionResponse(final Question question, final Choice answer) {
+        Validate.notNull(question);
+        Validate.notNull(answer);
+
+        if (!(question instanceof IsaacReorderQuestion)) {
+            throw new IllegalArgumentException(String.format(
+                    "This validator only works with IsaacReorderQuestion (%s is not ReorderQuestion)", question.getId()));
+        }
+
+        if (!(answer instanceof ItemChoice)) {
+            throw new IllegalArgumentException(String.format(
+                    "Expected ItemChoice for IsaacReorderQuestion: %s. Received (%s) ", question.getId(), answer.getClass()));
+        }
+
+        // These variables store the important features of the response we'll send.
+        Content feedback = null;                        // The feedback we send the user
+        boolean responseCorrect = false;                // Whether we're right or wrong
+
+        IsaacReorderQuestion reorderQuestion = (IsaacReorderQuestion) question;
+        ItemChoice submittedChoice = (ItemChoice) answer;
+
+        // STEP 0: Is it even possible to answer this question?
+
+        if (null == reorderQuestion.getChoices() || reorderQuestion.getChoices().isEmpty()) {
+            log.error("Question does not have any answers. " + question.getId() + " src: "
+                    + question.getCanonicalSourceFile());
+            feedback = new Content("This question does not have any correct answers");
```

Comment:
Very minor, but could this have punctuation at the end? I'd go with `!` but a full stop is also okay.